Sources/Core/Cache.swift

```diff
@@ -0,0 +1,113 @@
+public typealias Size = Int
+
+public protocol Cacheable {
+    func cacheSize() ->  Size
+}
+
+public protocol CacheProtocol {
```

Comment:
This name will clash with `Vapor.CacheProtocol` which could confuse people. This is a lower level cache and also doesn't support serializing for persistence. Maybe we could call it MemoryCache or SystemCache?